go/interpreter/lfvm/memory.go

```diff
@@ -176,47 +169,35 @@ func (m *Memory) Set(offset, size uint64, value []byte) error {
 			return errGasUintOverflow
 		}
 		if offset+size > m.Len() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.Len(), size, offset)
+			return errInsufficientMemory(m.Len(), size, offset)
 		}
 		copy(m.store[offset:offset+size], value)
 	}
 	return nil
 }
 
+func errInsufficientMemory(memSize, size, offset uint64) error {
```

Comment:
The naming for this function was commented before. 
I really like the technique, it yields clean code. 
To follow the function naming style, we can rename it to `makeInsuficientMemoryError`  
wdyt?.